src/components/GraphUtils.js

```diff
@@ -0,0 +1,202 @@
+
+  const transform = d3.zoomIdentity
+    .translate(width / 2 - scale * (graphX + graphWidth / 2), height / 2 - scale * (graphY + graphHeight / 2))
+    .scale(scale);
+
+  svg
+    .transition()
+    .duration(0)
+    .call(zoom.transform, transform);
+}
+
+export function filterByType(nodes, links, type) {
+  return nodes.filter(nodeGroup => nodeGroup.kind === type).filter(alert => {
+    var valid = false;
+    links.forEach(link => {
+      if(link.source === alert.id || link.target === alert.id) valid = true;
+    });
+    return !valid;
+  });
+}
+
+export function filterK8sNodesOtherNamespace(nodes, links) {
+  return nodes.filter(nodeGroup => nodeGroup.kind === 'node').filter(node => {
+    var valid = false;
+    const clusterNames = nodes.filter(nodeGroup => nodeGroup.kind === 'cluster').map(cluster => cluster.id);
+    const linksWithoutClusters = links.filter(link => !(clusterNames.includes(link.source) || clusterNames.includes(link.target)));
+
+    linksWithoutClusters.forEach(link => {
+      if(link.source === node.id || link.target === node.id) valid = true;
+    });
+    return !valid;
+  });
+}
+
+export function getNeighbours(links, object) {
+  const objectLinks = links.filter(link => link.source === object || link.target === object);
+  const neighbours = [];
+  objectLinks.forEach(link => {
+    if(link.source === object) {
+      neighbours.push(link.target);
+    } else {
+      neighbours.push(link.source);
+    }
+
+    return true;
+  });
+  return neighbours;
+}
+
+export function drag(simulation) {
+  function dragstarted(d) {
+    if (!d3.event.active) simulation.alphaTarget(0.3).restart();
+    d.fx = d.x;
+    d.fy = d.y;
+  }
+
+  function dragged(d) {
+    d.fx = d3.event.x;
+    d.fy = d3.event.y;
+  }
+
+  function dragended(d) {
+    if (!d3.event.active) simulation.alphaTarget(0);
+    d.fx = null;
+    d.fy = null;
+  }
+
+  return d3.drag()
+    .on('start', dragstarted)
+    .on('drag', dragged)
+    .on('end', dragended);
+}
+
+export function filterByNamespace(nodes, namespace) {
+  if(namespace) {
+    nodes = nodes.filter(nodeGroup => nodeGroup.properties.namespace === namespace || nodeGroup.kind === 'alert' ||
+      nodeGroup.kind === 'cluster' || nodeGroup.kind === 'node');
+  }
+
+  return nodes;
+}
+
+export function filterNodes(nodes, links, namespace) {
+  if(namespace !== '{}'){
+    const k8sNodesOtherNamespace = filterK8sNodesOtherNamespace(nodes, links);
+    const k8sNodesOtherNamespaceNames = k8sNodesOtherNamespace.map(node => node.id);
+    const k8sNodesOtherNamespaceLinks = links.filter(link => k8sNodesOtherNamespaceNames.includes(link.source) || k8sNodesOtherNamespaceNames.includes(link.target));
+
+    nodes = nodes.filter(nodeGroup => !k8sNodesOtherNamespace.includes(nodeGroup));
+    links = links.filter(linkGroup => !k8sNodesOtherNamespaceLinks.includes(linkGroup));
+  }
+
+  return [nodes, links];
+}
+
+export function filterByTypes(nodes, kinds) {
```

Comment:
`types` vs. `kinds`. Let's decide on one and use it consistently.

Reply:
Let's stay with ``kinds``, because it's parameter obtained from backend.